### ISSUE TITLE
Adapters versions should be semver. Patch number can differ from `dbt-core`

### ISSUE THREAD

**Issue by jtcohen6** [ClickHouse/dbt-clickhouse#63]
Hey team, thanks for your great work maintaining `dbt-clickhouse`!

It looks like you cut a release of `dbt-clickhouse` a week ago, [v1.1.0.1](https://github.com/ClickHouse/dbt-clickhouse/releases/tag/v1.1.0.1), which doesn't adhere to [semantic versioning](https://semver.org/). As of v1.0+, `dbt-core` expects installed adapter plugins to:
- be valid semantic versions
- match the `dbt-core` semantic version in its Major + Minor components — where the Patch number is allowed to differ

That logic lives [here](https://github.com/dbt-labs/dbt-core/blob/f8d347e5f82d52c39f0b0d7ac2e29ef8715fb10f/core/dbt/version.py#L127-L152O) in `dbt-core`.

Docs on this: https://docs.getdbt.com/docs/core-versions#how-dbt-core-uses-semantic-versioning

In short, I'd recommend that you make two changes:
- Loosen the `dbt-core` version dependency to be at the minor version level, instead of the patch level, by changing [this line](https://github.com/ClickHouse/dbt-clickhouse/blob/c2a6fe31f0949c494ffc0a08c325e65e9e83aa0a/setup.py#L56) to `dbt-core~={dbt_version}`
- Re-releasing v1.1.0.1 as v1.1.1

Let me or @dataders know if you have any questions!

**Comment by guykoh** [comment]
Thanks @jtcohen6, I followed your first suggestion.

**Comment by 1c7** [comment]
@guykoh  Hi, I think you should change the version using `x.y.z` too.
so `dbt` can work correctly

**Comment by 1c7** [comment]
Hi @jtcohen6, so far the latest version of `dbt-clickhouse` is still 1.1.0.2

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/1804755/174590088-3a45cfe4-39a4-4a93-be54-b1968010f149.png">

for unknown reason @guykoh don't want to make the version `valid semantic versions`. (something like `1.1.2`)

## My question to @jtcohen6 is: 
beside `dbt --version` wouldn't work, is there are any other side effect?
would version number contain 4 digit `x.y.z.k` cause other error make it unusable?

**Comment by 1c7** [comment]
## A little background (why we want to make `dbt-core` and `ClickHouse` work nicely together)
ClickHouse is open source, we are using it as analytic database(comebine with Superset).   
We want to use dbt with ClickHouse, so we can break long SQL code(more than 100 line) into multiple dbt model.   
we don't want to switch to Databrick/RedShift/Postgres/BigQuery just for dbt.

**Comment by guykoh** [comment]
Hi @1c7, sorry for the late reply, I will update the version to 1.1.2, there's no need to fork.

**Comment by 1c7** [comment]
@guykoh Thanks!
